app/models/anime.rb

```diff
@@ -111,10 +111,8 @@ def season_year
   alias_method :year_changed?, :season_changed?
 
   def update_unit_count_guess(guess)
-    return if episode_count
-    return if episode_count_guess && episode_count_guess > guess
+    return if episode_count || (episode_count_guess && episode_count_guess < guess)
```

Comment:
I think you got this backwards, look at the removed lines in the diff (they use `>` while this uses `<`)